src/storage-types/src/connections/aws.rs

```diff
@@ -136,58 +227,119 @@ impl AwsConfig {
+                let AwsAssumeRole {
+                    arn,
+                    session_name,
+                    mz_principal,
+                } = assume_role;
+                let assume_role_response = sts_client
+                    .assume_role()
+                    .role_arn(mz_principal.arn.clone())
+                    .role_session_name("materialize_assume_role")
+                    .send()
+                    .await?;
+
+                let temp_credentials = assume_role_response.credentials().ok_or_else(|| {
+                    anyhow!("Could not fetch temporary credentials with materialize role")
+                })?;
+
+                let access_key = temp_credentials
+                    .access_key_id()
+                    .ok_or_else(|| anyhow!("missing access_key"))?;
+                let secret_key = temp_credentials
+                    .secret_access_key()
+                    .ok_or_else(|| anyhow!("missing secret key"))?;
+
+                let credentials = aws_credential_types::Credentials::from_keys(
+                    access_key,
+                    secret_key,
+                    temp_credentials.session_token().map(|t| t.to_string()),
+                );
+
+                let session_name = session_name.to_owned().unwrap_or("materialize".to_string());
+                let mut role = AssumeRoleProvider::builder(arn).session_name(session_name);
+
+                if let Some(region) = &region {
+                    role = role.region(region.clone());
+                }
+
+                if let Some(external_id_suffix) = external_id_suffix {
+                    role = role.external_id(mz_principal.get_external_id(external_id_suffix));
+                }
+
+                SharedCredentialsProvider::new(role.build(credentials))
             }
-            cred_provider = SharedCredentialsProvider::new(role.build(cred_provider));
-        }
+        };
 
         let mut loader = aws_config::from_env()
             .region(region)
             .credentials_provider(cred_provider);
         if let Some(endpoint) = &self.endpoint {
             loader = loader.endpoint_url(endpoint);
         }
-        loader.load().await
+        Ok(loader.load().await)
     }
 
-    #[allow(clippy::unused_async)]
     pub(crate) async fn validate(
         &self,
-        _id: GlobalId,
-        _connection_context: &ConnectionContext,
+        id: GlobalId,
+        connection_context: &ConnectionContext,
     ) -> Result<(), anyhow::Error> {
-        Err(anyhow!("Validating SSH connections is not supported yet"))
+        match &self.auth {
+            AwsAuth::Credentials(_) => {
+                let aws_config = self
+                    .get_sdk_config(connection_context.secrets_reader.as_ref(), None)
+                    .await?;
+
+                let sts_client = aws_sdk_sts::Client::new(&aws_config);
+                let _ = sts_client.get_caller_identity().send().await?;
```

Comment:
These errors will be user facing. Are they descriptive enough to stand on their own, or do we need to massage them?